### PR TITLE
fix: do not eagerly connect on null provider

### PR DESCRIPTION
**Pull request by zzmp**

### CODE REVIEW

File: src/hooks/web3/index.tsx [hunk 1]
```diff
@@ -72,18 +72,26 @@ export function Provider({
...
     connectors.network.activate().catch(() => undefined)
-  }, [connectors.metaMask, connectors.network, connectors.user, connectors.walletConnect])
+  }, [
+    connectors.metaMask,
+    connectors.network,
+    connectors.user,
+    connectors.walletConnect,
+    provider,
+    shouldEagerlyConnect,
+    web3ReactConnectors.user,
```
Comment (cmcewen) [inline src/hooks/web3/index.tsx:93]: why should this trigger the effect?

File: src/hooks/web3/index.tsx [hunk 2]
```diff
@@ -72,18 +72,26 @@ export function Provider({
...
     }
     connectors.network.activate().catch(() => undefined)
-  }, [connectors.metaMask, connectors.network, connectors.user, connectors.walletConnect])
+  }, [
+    connectors.metaMask,
+    connectors.network,
+    connectors.user,
+    connectors.walletConnect,
+    provider,
+    shouldEagerlyConnect,
```
Comment (cmcewen) [inline src/hooks/web3/index.tsx:92]: this will always change when provider changes right?

Comment (zzmp) [reply src/hooks/web3/index.tsx:92]: yes. provider should be omitted.

File: src/hooks/web3/index.tsx (hunk 1 above)
Comment (zzmp) [reply src/hooks/web3/index.tsx:93]: this should be omitted.

**Review by cmcewen — APPROVED** [review]

**Review by cmcewen — APPROVED** [review]

**Pull request merged**